src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -371,4 +374,29 @@ private void ApplyDaoTransition(Block block)
             }
         }
     }
+
+    private class TxHashCalculator(List<Block> suggestedBlocks) : IThreadPoolWorkItem
+    {
+        public static void CalculateInBackground(List<Block> suggestedBlocks)
+        {
+            // Memory has been reserved on the transactions to delay calculate the hashes
+            // We calculate the hashes in the background to release that memory
+            ThreadPool.UnsafeQueueUserWorkItem(new TxHashCalculator(suggestedBlocks), preferLocal: false);
+        }
+
+        void IThreadPoolWorkItem.Execute()
+        {
+            // Hashes will be required for PersistentReceiptStorage in UpdateMainChain ForkchoiceUpdatedHandler
+            // Which occurs after the block has been processed; however the block is stored in cache and picked up
+            // from there so we can calculate the hashes now for that later use.
+            foreach (Block block in CollectionsMarshal.AsSpan(suggestedBlocks))
+            {
+                foreach (Transaction tx in block.Transactions)
+                {
+                    // Calculate the hashes to release the memory from the transactionSequence
+                    tx.CalculateHashInternal();
```

Comment:
Can we release memory without calculating the hashes if we don't need to?

Reply:
We need the hashes for forkchoice; this block gets stored in cache, then when forkchoice comes along it expects the hashes for `Receipts.PersistentReceiptStorage.EnsureCanonical`